src/app/products/manage-tax-components/create-tax-component/create-tax-component.component.html

```diff
@@ -0,0 +1,96 @@
+<div class="container">
+
+  <mat-card>
+
+    <form [formGroup]="taxComponentForm" (ngSubmit)="submit()">
+
+      <mat-card-content>
+
+        <div fxLayout="column">
+
+          <mat-form-field>
+            <mat-label>Name</mat-label>
+            <input matInput required formControlName="name">
+            <mat-error *ngIf="taxComponentForm.controls.name.hasError('required')">
+              Name is <strong>required</strong>
+            </mat-error>
+          </mat-form-field>
+
+          <mat-form-field>
+            <mat-label>Percentage</mat-label>
+            <input type="number" matInput required formControlName="percentage">
+            <mat-error *ngIf="taxComponentForm.controls.percentage.hasError('required')">
+              Percentage is <strong>required</strong>
+            </mat-error>
+            <mat-error *ngIf="taxComponentForm.controls.percentage.hasError('min') ||
+            taxComponentForm.controls.percentage.hasError('max')">
+              Percentage <strong>should</strong> be between one and a hundred
```

Comment:
```suggestion
              Percentage <strong>should</strong> be between 1 and 100
```

Reply:
@abhaychawla I was just wondering, wouldn't it be better if the percentage have to be larger than 0 and at most 100, so that the user will be able to enter decimal tax values such as 0.4% etc.

Reply:
If that is the case, he can simply reuse the regex from the create-tax file to do this too, as angular doesn't provide a larger than validator.

Reply:
Alright, @punwai I checked. @OussEmaDevCode it seems decimal values are allowed by the API. Let us use a `regex` pattern of decimal values greater than 0 instead of min and max validators and update the error message as he specified above.

Reply:
@OussEmaDevCode Could I suggest that for your mental health, you use the same regex to validate for percentages larger than 0, but then still use a max to validate for it to be at most 100. It will be a very complicated regex to deal with both inequalities.

Reply:
Good suggestion !!